backend/views/syson-tree-explorer-view/src/test/java/org/eclipse/syson/tree/explorer/view/services/SysONDefaultExplorerServicesTest.java

```diff
@@ -160,9 +162,82 @@ public boolean test(Object arg0) {
 
         IExplorerServices explorerServices = new ExplorerServices(objectService, labelService, List.of(), representationMetadataSearchService, readOnlyObjectPredicate, defaultObjectSearchService);
 
-        ISysONExplorerFilterService filterService = new SysONExplorerFilterService(this.sysONResourceService);
+        ISysONExplorerFilterService filterService = new SysONExplorerFilterService(sysONResourceService);
 
-        this.sysONDefaultExplorerServices = new SysONDefaultExplorerServices(identityService, contentService, representationMetadataSearchService, explorerServices, labelService, filterService, this.sysONResourceService);
+        return new SysONDefaultExplorerServices(identityService, contentService, representationMetadataSearchService, explorerServices, labelService, filterService, sysONResourceService);
+    }
+
+    /**
+     * Unit tests for
+     * {@link ISysONDefaultExplorerService#hasChildren(Object, org.eclipse.sirius.components.core.api.IEditingContext, List, List, List)}.
+     * 
+     * @author flatombe
+     * @author dvojtise
+     */
+    @Nested
+    @DisplayName("boolean hasChildren(Object, IEditingContext, List<RepresentationMetadata>, List<String>, List<String>)")
+    public class TestHasChildren {
```

Comment:
There is no need for @Nested and sub classes.
Please add all tests at the root of the file.